src/schema/uuid/thread/utils.ts

```diff
@@ -74,3 +74,11 @@ export function createThreadResolvers(): ThreadAwareResolvers {
     },
   }
 }
+
+export function encodeThreadId(firstCommentId: number) {
+  return Buffer.from(`t${firstCommentId}`).toString('base64')
+}
+
+export function decodeThreadId(threadId: string) {
+  return parseInt(Buffer.from(threadId, 'base64').toString('ascii').substr(1))
+}
```

Comment:
```suggestion
export function decodeThreadId(threadId: string): number | null {
  const result = parseInt(Buffer.from(threadId, 'base64').toString('ascii').substr(1))
  return Number.isNaN(result) || result <= 0 ? null : result
}
```

With this one cannot forget to check, whether a passed id is a valid id or not since the typescript compiler will complain when `number | null` is used in a case where `number` is necessary. Example:

```ts
const threadId = "..."
const id = decodeThreadId(threadId)

// will complain since id = null is not allowed, 
// so you don't forget to handle the case where `threadId` is not valid
await dataSources.serlo.setThreadArchive({ id }) 
```

Reply:
The option monad might be useful here: https://dev.to/waynevanson/fp-ts-meets-if-else-and-switch-part-1-3-2lpf

Reply:
Did it with the suggested `number | null` for now. Let's talk about it.